lib/os.js

```diff
@@ -60,28 +60,53 @@ const {
   setPriority: _setPriority,
 } = internalBinding('os');
 
-function getCheckedFunction(fn) {
-  return hideStackFrames(function checkError(...args) {
-    const ctx = {};
-    const ret = fn(...args, ctx);
-    if (ret === undefined) {
-      throw new ERR_SYSTEM_ERROR(ctx);
-    }
-    return ret;
-  });
-}
-
 const {
   0: type,
   1: version,
   2: release,
   3: machine,
 } = _getOSInformation();
 
-const getHomeDirectory = getCheckedFunction(_getHomeDirectory);
-const getHostname = getCheckedFunction(_getHostname);
-const getInterfaceAddresses = getCheckedFunction(_getInterfaceAddresses);
-const getUptime = getCheckedFunction(_getUptime);
+let homedirCached = null;
```

Comment:
I don't think it's worth the potential issues that could arise from caching this, especially since this is not generally going to be a hot code path.